src/main/resources/assets/js/components/AppModalComponent.jsx

```diff
@@ -31,17 +33,49 @@ var AppModalComponent = React.createClass({
     rollBackApp: React.PropTypes.func.isRequired,
     scaleApp: React.PropTypes.func.isRequired,
     suspendApp: React.PropTypes.func.isRequired,
-    tasksFetchState: React.PropTypes.number.isRequired
+    tasksFetchState: React.PropTypes.number.isRequired,
+    router: React.PropTypes.object.isRequired
   },
 
   getInitialState: function () {
+    var appid = this.props.model.get("id");
+    var activeTabId;
+
+    var tabs = _.reduce(tabsTemplate, function (current, tab) {
```

Comment:
Yes, but need unique ids at the moment for the NavTabComponent.
Should be a seperate refactoring task.